src/main/entities/NewPassword.java

```diff
@@ -0,0 +1,146 @@
+package entities;
+
+public class NewPassword extends Password {
+
+    private int passlen;
+    private String choices;
+
+    public void setPasslen(int passlen) {
+        this.passlen = passlen;
+    }
+
+    public void setChoices(String choices) {
+        this.choices = choices;
+    }
+
+    public void generatePassword() {
+        if (choices.length() == 4) {
```

Comment:
The 4 and 3 and 2 and "L" and so on are magic values, especially the letters. Maybe create constants for them?